vernac/vernacentries.ml

```diff
@@ -1329,6 +1329,32 @@ let vernac_begin_segment ~interactive f =
   }
 [@@ocaml.warning "-40"]
 
+(* External dependencies *)
+
+let vernac_extra_dep ?loc from file id =
+  if Global.sections_are_opened () then
+    user_err ?loc Pp.(str "Extra Dependencies cannot be declared inside sections.");
+  let hd, tl = Libnames.repr_qualid from in
+  let from = Libnames.add_dirpath_suffix hd tl in
+  ComExtraDeps.declare_extra_dep ?loc ~from ~file id
+
+let constrexpr_is_string s = function
+  | { CAst.v = Constrexpr.CRef(x,None) } -> string_of_qualid x = s
+  | _ -> false
+
+let forward_compat_extra_dependency ?loc = function
+  | [ CommentConstr from_str;
+      CommentConstr { CAst.v = Constrexpr.CRef(from,None) };
+      CommentConstr extra_str;
+      CommentConstr dependency_str;
+      CommentString file ] when
+        constrexpr_is_string "From" from_str&&
+        constrexpr_is_string "Extra" extra_str &&
+        constrexpr_is_string "Dependency" dependency_str
+      ->
+      vernac_extra_dep ?loc from file None
```

Comment:
Deprecation warning? I don't think we want to keep this forever.

Reply:
done